src/daft-logical-plan/src/optimization/rules/enrich_with_stats.rs

```diff
@@ -0,0 +1,26 @@
+#[derive(Default, Debug)]
+pub struct EnrichWithStats {}
+
+impl EnrichWithStats {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+use std::sync::Arc;
+
+use common_error::DaftResult;
+use common_treenode::{Transformed, TreeNode};
+
+use super::OptimizerRule;
+use crate::LogicalPlan;
+
+// Add stats to all logical plan nodes in a bottom up fashion.
```

Comment:
add a note that this requires all scan nodes to be materialized prior to running this rule

Reply:
Sounds good, done